src/types/inc/utils.hpp

```diff
@@ -52,6 +52,20 @@ namespace Microsoft::Console::Utils
     bool StringToUint(const std::wstring_view wstr, unsigned int& value);
     std::vector<std::wstring_view> SplitString(const std::wstring_view wstr, const wchar_t delimiter) noexcept;
 
+    enum PasteOption
+    {
+        // Convert Windows-space \r\n and \n line-endings to \r line-endings.
+        CarriageReturnNewline = 1u << 0,
+        // For security reasons, remove most control characters
+        FilterControlCodes = 1u << 1,
+        // Bracketed paste. See: https://www.xfree86.org/current/ctlseqs.html#Bracketed%20Paste%20Mode.
+        Bracketed = 1u << 2

```

Comment:
This part may not be properly extensible for the Conhost part of bracketed paste. The idea there is to introduce a new type of input event that maps to a higher level concept of a "Clipboard Paste". It will, when coupled with the right console modes, produce the escape sequences for bracketing. It won't be a single string transformation like this one is. :smile:

Reply:
Here's my thinking.

If we remove bracketing from the paste options and leave it in TerminalCore, we can use this for filtering control sequences and newlines only. Simplifying the logic here is a bonus!

Reply:
That's actually brilliant! I was so caught up in the idea that bracketing needs to be coupled with filtering control codes. But it feels more intuitive & clean to separate them. This will simplify the logic & provides possibility for conhost to support bracketing.